topics/07_web_ui_testing/main_web_automation_with_playwright.md

```diff
@@ -1,3 +1,383 @@
 # Topic 7: Web Automation with Playwright
 
-Coming soon.
+## Basic approach of Web Automation
+- DOM, 
+- DevTool,
+- Locators
+- Selenium
+
+### DOM
+DOM (Document Object Model) 
+The document currently loaded in each one of your browser tabs is represented by a document object model. This is a "tree structure" representation created by the browser that enables the HTML structure to be easily accessed by programming languages.
+
+#### Links to read about DOM
+- [What is DOM](https://developer.mozilla.org/en-US/docs/Web/API/Document_Object_Model/Introduction)
+- [DOM Manipulating_documents](https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Client-side_web_APIs/Manipulating_documents)
+- [DOM introduction](https://developer.mozilla.org/en-US/docs/Web/API/Document_Object_Model/Introduction)
+
+### DevTool
+DevTools (short for Developer Tools) are built-in utilities in web browsers that allow developers to inspect, debug, and optimize their web applications directly from the browser. They provide a range of powerful features that facilitate front-end development, testing, and performance analysis, making it easier to understand how web pages are constructed and behave.
+It is a part of many browsers particularly in Chrome, Firefox, Safari, and Edge
+
+It incorporates several panels
+For Web Automation the most important is Elements Panel:
+It contains DOM
+It allows viewing and edit the HTML structure and CSS styles of a webpage in real-time. You can change attributes, add styles, and see how these affect the page immediately.
+Layout and Box Model: Visualize elements’ layout properties like padding, margin, border, and dimensions.
+
+### Locators
+Locators are used to identify and interact with WebElements within a web page’s Document Object Model (DOM). They serve as the foundation for automating web application tests by allowing testers to perform actions such as clicking buttons, entering text, or verifying content.
+In Python locators are objects that are created by using /consuming string that define the place of element within DOM
+
+#### The most typical types of Locators:
```

Comment:
Add practical examples for each locator you are mentioning. You can ask for examples from chatGPT. You need to provide a good prompt to have good examples. 

Would be more interesting to have things like this:

"""
##### ID Locator
Targets elements with a unique id attribute. It's the fastest and most reliable locator if **available.**

In Selenium:
```python
from selenium import webdriver

driver = webdriver.Chrome()
driver.get("https://example.com")
element = driver.find_element("id", "uniqueId")
element.click()
driver.quit()
```

In Playwright:
```python
from playwright.sync_api import sync_playwright

with sync_playwright() as p:
    browser = p.chromium.launch()
    page = browser.new_page()
    page.goto("https://example.com")
    element = page.locator("#uniqueId")
    element.click()
    browser.close()
```

##### Class Name Locator
Matches elements by their class attribute, often used for elements styled similarly.

In Selenium:
```python
element = driver.find_element("class name", "button-class")
element.click()
```

In Playwright:
```python
element = page.locator(".button-class")
element.click()
```

and so on..

Reply:
Done.